### ISSUE TITLE
Create exact error message for z.array().length().

### ISSUE THREAD

**Issue by john-schmitz** [colinhacks/zod#1611]
Given this schema:

`const testSchema = z.array(z.number()).length(4);`

For arrays with less than 4 elements, it returns the message: 
>Should have at least 4 items

For arrays with more than 4 elements, it returns the message: 
>Should have at most 4 items

I wish instead for it to return a message explaining that it should be precisely 4 items:
>  Should have exactly 4 items

I know I could achieve that with a custom refine validation, but it would be nice to have it baked in.

Of course, if you find this suggestion valid, I could make a PR for it.

**Comment by maxArturo** [comment]
Hi @john-schmitz ! I'll provide my opinion based on what I see implemented. Here's the def for `z.array().length()`

```ts
  length(len: number, message?: errorUtil.ErrMessage): this {
    return this.min(len, message).max(len, message) as any;
  }
```

That's why you see the behavior described in your issue - `min` gets validated before `max`, and their respective messages get shown in the validation result. I believe building an `exactly` out of `min` and `max` is probably the way to do it (keeping in with the don't-repeat-yourself ethos). 

_However_, there's an opportunity to build a more contextually-correct error message precisely [in that line](https://github.com/colinhacks/zod/blob/2c7793821c3245ca181a2f16fd06fc0cea7e9a4e/src/types.ts#L1625) that gives a more contextually-correct error message, if you'd like to submit a PR.

**Comment by maxArturo** [comment]
Oh a final note - you don't need a custom refine validation! It's as easy as this

```ts
  const testSchema = z
    .array(z.number())
    .length(4, { message: "needs to be exactly 4" });

  testSchema.parse([]);
``` 
In case that saves you any trouble.

**Comment by john-schmitz** [comment]
fixed with pr #1620
